lib/construct.ts

```diff
@@ -428,15 +428,16 @@ export class Node {
    * Invokes "prepare" on all constructs (depth-first, post-order) in the tree under `node`.
    */
   public prepare() {
-    const constructs = this.findAll(ConstructOrder.PREORDER);
+    let constructs = this.findAll(ConstructOrder.PREORDER);
```

Comment:
I think I would maybe just inline the `findAll` into the `for` loop. Do keep the comment saying why we need to do 2 queries, in case someone gets the sense to combine them again.

Reply:
+1 to this suggestion.

Reply:
done